src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,18 +1,29 @@
 import NextAuth from 'next-auth/next';
 import type { NextAuthOptions } from 'next-auth';
+import { Provider } from 'next-auth/providers';
 import GoogleProvider from 'next-auth/providers/google';
+import GitHubProvider from 'next-auth/providers/github';
 
 const GOOGLE_ID = process.env.GOOGLE_ID ?? '';
 const GOOGLE_SECRET = process.env.GOOGLE_SECRET ?? '';
 
-const options: NextAuthOptions = {
-  providers: [
-    GoogleProvider({
-      clientId: GOOGLE_ID,
-      clientSecret: GOOGLE_SECRET,
-    }),
-  ],
-};
+let providers: Provider[] = [
+  GoogleProvider({
+    clientId: GOOGLE_ID,
+    clientSecret: GOOGLE_SECRET,
+  }),
+];
+
+if (process.env.ENABLE_GITHUB_AUTH === 'true') {
+  providers.push(
+    GitHubProvider({
+      clientId: process.env.GITHUB_ID ?? '',
+      clientSecret: process.env.GITHUB_SECRET ?? '',
+    })
+  );
+}
```

Comment:
I needed this because it was difficult to test auth locally and didn't wanted to go with setting up google auth, since I already had GH credentials + since this is hidden behind feature flag, it's won't be exposed in prod